practice-app/api/views.py

```diff
@@ -1,3 +1,191 @@
-from django.shortcuts import render
+from django.http import Http404
+from django.http import HttpResponse, JsonResponse
 
-# Create your views here.
+from rest_framework import status
```

Comment:
You have imported JsonResponse, JSONRenderer, BrowsableAPIRenderer, APIView, action, JSONParser but didn't use them in the code. Maybe it would simplify the imports if you delete them. Other than that views.py look okay and it includes all functionalities including POST, GET, PUT, DELETE.

Reply:
Unnecessary imports are deleted in the next commit.